src/framework/linalg/linops/linops_unordered_map.hpp

```diff
@@ -0,0 +1,194 @@
+/**
+ * This code is part of Qiskit.
+ *
+ * (C) Copyright IBM 2018, 2019.
+ *
+ * This code is licensed under the Apache License, Version 2.0. You may
+ * obtain a copy of this license in the LICENSE.txt file in the root directory
+ * of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Any modifications or derivative works of this code must retain this
+ * copyright notice, and modified files need to carry a notice indicating
+ * that they have been altered from the originals.
+ */
+
+#ifndef _aer_framework_linalg_linops_unordered_map_hpp_
+#define _aer_framework_linalg_linops_unordered_map_hpp_
+
+#include <functional>
+#include <unordered_map>
+
+#include "framework/linalg/almost_equal.hpp"
+#include "framework/linalg/enable_if_numeric.hpp"
+namespace AER {
+namespace Linalg {
+
+// This defines functions add, sub, mul, div and iadd, imul, isub, idiv
+// that work for numeric unordered_map types
+
+//----------------------------------------------------------------------------
+// Linear operations
+//----------------------------------------------------------------------------
+template <class T1, class T2, class T3, class T4, class T5,
+          typename = enable_if_numeric<T2>>
+std::unordered_map<T1, T2, T3, T4, T5> add(
```

Comment:
Reminder: you can use `decltype(auto)` as a return type here as well :)

Reply:
Good to know